_posts/2018-08-30-pair-programming.markdown

```diff
@@ -0,0 +1,184 @@
+---
+layout: epic
+title: "The Hows and Whys of Pair Programming"
+date: 2018-08-30
+author: [yuki, orta]
+categories: [pairing, culture]
+---
+
+Why pair program? As new engineers join Artsy we've been experimenting with different programming cultures - Yuki
+came from Pivotal Labs where they have a strong pair programming culture and introduced it at Artsy - it's been
+about a year and a half and we're all really loving the changes he's introduced.
+
+I asked Yuki if he'd pair program with me on a blog, post on Pair Programming, and this is it. So, we're going to
+dive into what pair programming is, why you should do it, what are good mental models to think about, the techniques
+you can use to make it work, what hardware you might need and how Yuki persuaded so many of us to start doing it
+more often.
+
+<!-- more -->
+
+## Why pair program?
+
+The idea of pair programming came from a very popular book called [Extreme Programming Explained: Embrace
+Change][extreme-explained]. Code review has been a good practice in software development. Many organizations have
+adopted it, but XP (Extreme Programming) literally takes this **more extremely** - what if the engineer sitting next
+to you is reviewing your code as soon as you write it? This requires two engineers to simultaneously pay attention
+to the code that's being written, discuss implementation details, and sometimes socialize. That's how pair
+programming was born.
+
+Pair programming seems as easy as it sounds, and to some extent that's true. However, with just a few key points
+you'll be able to make your pairing session much more valuable.
+
+To think about when pairing is most valuable, let's think about code review first. Modern software organizations
+have implemented a process where engineers send pull requests to get reviews on GitHub (or whatever tool your
+organization uses for reviewing code). This is a very powerful process, but most of us have probably experienced a
+situation where a pull request with a very simple change created a very long discussion. Sometimes
+that's because of your lack of context in the work you are doing. Sometimes it's because you are working on a
+system you are unfamiliar with. Your co-worker left a lot of comments on your pull request, and you had to push a lot
+of extra commits, or even re-wrote the entire pull request. That's when pairing becomes very valuable to your team.
+
+Another good example is on-boarding new hires. Coming from an agile consultancy and a photo-product company, I was
+nowhere close to an Art expert when I joined Artsy. This was the time I wanted to question not just technical
+questions, but also very basic questions about art. How do people find and purchase artworks? Who actually sells
+artworks? What is an art fair? And auctions? Christina Thompson, on the other hand, has been at Artsy for more than
```

Comment:
`Christina Thompson` could be a link to @xtina-starr's GitHub/Twitter, if she'd like.

Reply:
@xtina-starr would you mind if I add a link to https://medium.com/artsy-blog/what-it-feels-like-to-work-in-a-supportive-environment-for-female-engineers-3c994a001007?